simpa/utils/libraries/tissue_library.py

```diff
@@ -68,14 +70,24 @@ def muscle(self, background_oxy=None, blood_volume_fraction=None):
         custom_water.f_ray = OpticalTissueProperties.FRAY_MUSCLE_TISSUE
 
         # generate the tissue dictionary
-        return (MolecularCompositionGenerator()
-                .append(MOLECULE_LIBRARY.oxyhemoglobin(fraction_oxy))
-                .append(MOLECULE_LIBRARY.deoxyhemoglobin(fraction_deoxy))
-                .append(value=MOLECULE_LIBRARY.muscle_scatterer(
+        if type(blood_volume_fraction) == (int, float, np.int64, np.float64) or blood_volume_fraction is None:
+            return (MolecularCompositionGenerator()
+                    .append(MOLECULE_LIBRARY.oxyhemoglobin(fraction_oxy))
+                    .append(MOLECULE_LIBRARY.deoxyhemoglobin(fraction_deoxy))
+                    .append(value=MOLECULE_LIBRARY.muscle_scatterer(
                         volume_fraction=1 - fraction_oxy - fraction_deoxy - water_volume_fraction),
-                        key="muscle_scatterers")
+                key="muscle_scatterers")
                 .append(custom_water)
                 .get_molecular_composition(SegmentationClasses.MUSCLE))
+        else:
+            return (MolecularCompositionGenerator()
+                    .append(MOLECULE_LIBRARY.oxyhemoglobin(fraction_oxy))
+                    .append(MOLECULE_LIBRARY.deoxyhemoglobin(fraction_deoxy))
+                    .append(value=MOLECULE_LIBRARY.muscle_scatterer(
+                        volume_fraction=1 - fraction_oxy - fraction_deoxy - water_volume_fraction),
+                key="muscle_scatterers")
+                .append_filler(custom_water)
+                .get_molecular_composition(SegmentationClasses.MUSCLE))
```

Comment:
Sorry for another question...: What was the reasoning behind not having a filler molecule by default?
This could also be simplified by extracting the decorator functions that are the same and then only add the filler if it is not scalar. But I think it it fine the way it is, too.

Reply:
This I did purely for efficiency. It seemed unnecessary to be using filler functions generally as it requires passing tensors rather than just single numbers, which could be costly when checking things such as that the volume is full. In reality, the change is marginal (sometimes suggesting that filler by default is quicker). See the stats below for using filler by default.

![Screenshot from 2024-06-24 10-07-23](https://github.com/IMSY-DKFZ/simpa/assets/155451092/beaa2ea1-a230-43c5-b453-4124d19330eb)

Memory here however does clearly increase a lot, this number is indeed quite unpredictable but even still the orders here are clearly considerably higher.

Reply:
Also, I've been thinking, is append filler required here at all? It seems that it leads to us doing some computation twice (calculating the filler volume for example will be done again in molecule_library) and therefore it might best to have it that here we exclusively use .append

If not, then I agree I should neaten up the code to avoid doubling up lines unnecessarily